homeassistant/components/intellifire/number.py

```diff
@@ -25,21 +25,66 @@ async def async_setup_entry(
     """Set up the fans."""
     coordinator: IntellifireDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
 
-    description = NumberEntityDescription(
-        key="flame_control",
-        name="Flame control",
-        icon="mdi:arrow-expand-vertical",
-    )
-
     async_add_entities(
         [
             IntellifireFlameControlEntity(
-                coordinator=coordinator, description=description
-            )
+                coordinator=coordinator,
+                description=NumberEntityDescription(
+                    key="flame_control",
+                    name="Flame control",
+                    icon="mdi:arrow-expand-vertical",
+                ),
+            ),
+            IntellifireTimerControlEntity(
+                coordinator=coordinator,
+                description=NumberEntityDescription(
+                    key="sleep_control",
+                    name="Sleep timer",
+                    icon="mdi:bed-clock",
+                ),
+            ),
```

Comment:
The `NumberEntityDescription` should be defined as two constants, not generated inside `async_setup_entry`.